msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase532736.java

```diff
@@ -0,0 +1,116 @@
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+// MSAL Falls Back on WebView When All Browsers are Disabled
+// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/532736
+public class TestCase532736 extends AbstractMsalUiTest {
+
+    @Test
+    public void test_532736() throws Throwable {
+        final String username = mLabAccount.getUsername();
+
+        // Disable Chrome
+        ISettings settings = getSettingsScreen();
+        BrowserChrome chrome = ((BrowserChrome) mBrowser);
+
+        settings.disableAppThroughSettings(chrome.getPackageName());
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                // Assert that webview UI object exists
+                final UiObject webView = UiAutomatorUtils.obtainUiObjectWithUiSelector(new UiSelector().className(WebView.class), CommonUtils.FIND_UI_ELEMENT_TIMEOUT);
+                Assert.assertTrue(webView.exists());
+
+                // Assert that no objects from Chrome package are present
+                final UiObject chromeObject = UiAutomatorUtils.obtainUiObjectWithUiSelector(new UiSelector().packageName("com.android.chrome"), CommonUtils.FIND_UI_ELEMENT_TIMEOUT);
+                Assert.assertFalse(chromeObject.exists());
+
+                // Some cleanup after the test passes
+                settings.enableAppThroughSettings(chrome.getPackageName());
+            }
+        }, TokenRequestTimeout.SILENT);
+    }
+
+    @Override
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
+    }
+
+    @Override
+    public TempUserType getTempUserType() {
+        return null;
+    }
+
+    @Override
+    public String[] getScopes() {
+        return new String[]{"User.read"};
+    }
+
+    @Override
+    public String getAuthority() {
+        return mApplication.getConfiguration().getDefaultAuthority().getAuthorityURL().toString();
+    }
+    @Override
+    public int getConfigFileResourceId() {
+        return R.raw.msal_config_browser;
```

Comment:
Can we also add a similar test case when config is `default`? We don't have to do this right now, but just something to think about from overall testing effort point of view

Reply:
Is the expected behavior the same?

Reply:
for default it tries in this order
CustomTab, Browser, WebView

Reply:
Done

Reply:
Added a parameterized test to this case to also run it on default config. I did not change anything in the test steps, just which config file is used. both are passing.